ios/RNSScreenStack.mm

```diff
@@ -602,11 +608,18 @@ - (void)setPushViewControllers:(NSArray<UIViewController *> *)controllers
       // no animation and do animated push of the last item
       NSMutableArray *newControllers = [NSMutableArray arrayWithArray:controllers];
       [newControllers removeLastObject];
-      [_controller setViewControllers:newControllers animated:NO];
+
+      // when array with current view controllers is equal to array with new ones,
+      // we shouldn't try to set push view controllers, since it may result with error
+      // about trying to push the same view controller more than once. This may be the case
+      // when we're trying to navigate to the one screen and navigate to the other one in the same time.
 #ifdef RCT_NEW_ARCH_ENABLED
```

Comment:
We should move the #ifdef before the comment I think

Reply:
Resolved in https://github.com/software-mansion/react-native-screens/pull/2069/commits/dbddc64d8d25567087fea904206626187b2c0001.